lib/spack/spack/main.py

```diff
@@ -528,6 +528,8 @@ def __call__(self, *argv, **kwargs):
 
         Keyword Args:
             fail_on_error (optional bool): Don't raise an exception on error
+            prepend_flags (optional list): List of flags to add before command:
+                simulates spack [prepend_flags] [command] [*argv]
```

Comment:
The fix is fine, but I don't think we should be adding args to business code that are relevant only for testing purposes

Reply:
Do you have suggestions? I'm not too familiar with the code base